### DISCUSSION TITLE
비회원도 사용할 수 있는 기능 + 로그인 플로우 개선

### DISCUSSION THREAD

**Discussion by Jaymyong66** [woowacourse-teams/2024-code-zap#465] [이야기해봐요]
다소 PM스러운 고민이긴 하지만..

## 비회원도 사용할 수 있는 기능
서비스를 이용하기 위해 회원가입은 생각보다 큰 장벽이다.
특히 우리는 현재 소셜 가입도 되지 않는다.

비회원도 둘러볼 수 있는 페이지나 기능이 있어야하고, 이후 회원가입을 결정하도록 하면 좋지 않을까?
게다가 런칭데이다! 많은 유저가 쉽게 사용해 볼 수 있어야 한다고 생각한다.

무슨 기능을 넣어볼 수 있을까?
(ex. 구경가기, 코멘트, 좋아요..)

[서비스 이용 시, 회원가입이 필수여야 할까?](https://brunch.co.kr/@jiwoomaster/23)

## 로그인 플로우 개선
[구경가기] 기능이 생긴다고 가정하고, 내 템플릿 조회나 템플릿 업로드 페이지는 회원만 가능한 기능일 것이다.
비회원이 이를 접근했을 때, 현재는 토스트 메시지로 "로그인을 해주세요"를 띄우고 다짜고짜 로그인 페이지로 리다이렉트 시킨다.
<img width="1126" alt="스크린샷 2024-08-17 오후 2 07 20" src="https://github.com/user-attachments/assets/c96d669c-2e0f-438e-a2a4-30454cb80098">

다른 서비스에서는 해당 기능을 사용하려면 왜 로그인을 해야하는지의 이유도 함께 안내해주어, **사용자가 가입을 해야하는 이유를 명확히 인지하도록 한다.**

또 로그인 시 수집되는 정보를 간략히 알려주기도 한다.

<img width="366" alt="스크린샷 2024-08-17 오후 2 02 33" src="https://github.com/user-attachments/assets/f21f6dae-b1ec-486b-9995-5c9189315ca7">
<img width="159" alt="스크린샷 2024-08-17 오후 2 02 57" src="https://github.com/user-attachments/assets/e82a9941-2b5c-42d7-93f9-33f300b39c28">

위와 같이 새로운 페이지를 만드는 것이 부담스럽다면 모달도 괜찮을 것 같다.

<img width="153" alt="스크린샷 2024-08-17 오후 2 03 44" src="https://github.com/user-attachments/assets/0bc2aee7-dc09-475a-abff-019a94e4a8f6">

**Comment by kyum-q** [comment]
추가적으로 어떻게 하면 사람들의 흥미를 끌 수 있을까에 대해 고민해봤는데, 리뷰나 코치들 조언을 넣으면 좋을 것 같아요

코드관련 서비스인 만큼 의견이 쟁쟁한 것들을 같은 태그?로 만들어놓는거죠
이러면 좋아요 기능이 있으면 확실히 좋을 것 같아요

---

EX. 쟁쟁한 의견 예시: 정적 팩토리 메서드 사용해야 하나?

- 1번 템플릿
   - 제목: 정적 팩토리 메서드 왜 써?
   - 설명: ~~한 이유로 쓰지마
   - 코드: 정팩메 사용 안한 코드 (더 좋은 방안을 보여준다)
   - 태그: 정팩메, VS, 반대
- 2번 템플릿
   - 제목: 정적 팩토리 안쓰면 생성자가 너무 복잡해
   - 설명: ~~한 이유로 안좋을 게 없고 편하다
   - 코드: 정팩메 사요 코드
   - 태그: 정팩메, VS, 찬성

EX. 코치님의 코드 받는 예시: (네오) 테스트 사용 시 var 사용한다
   - 제목: var 사용해봤어?
   - 설명: 테스트에서 var 사용하면 ~~한 이유로
   - 코드: 네오의 코드
   - 태그: 나코친데

---

예시를 대강 들어봤는데 저라면 이런 코드들이 많다면 흥미가 훨씬 생길 것 같아요😁

+ 그래서 슬슬 개발 발전만큼 데이터를 넣는 것에 시간을 쓰면 좋을 것 같습니자

**Comment by Jaymyong66** [comment]
+) 우려되는 부분으로 본래 만두 페르소나로써 필요한 기능이 아닐 수 있음을 염두 해야할 것 같습니다.

**Reply by kyum-q** [reply]
만두를 페르소나로 정하고 구현해야하는 기능은 거의 완성 됬다고 생각해요~ 물론 마지막 QA 를 더 꼼꼼하게 하긴 하겠지만~~
그러니 지금 기능이 동작 잘하는지만 하고 페르소나를 하나 더 만들거나 해서 해당 기능 하는건 좋다고 생각합니다~

**Reply by Jaymyong66** [reply]
너.무.좋.습.니.다. 켬.미.해.줘.

**Comment by zangsu** [comment]
## 이거는 하자고 제안하는 게 아니라 아이디어 공유라서 가볍게만 읽어주시거나 넘어가도 됩니다!!!

어제 옆 조의 모 크루가 내 주었던 아이디어인데요
똑같은 기능을 우리의 서버로 이용하는 대신, 로컬 스토리지(?) 로 이용할 수 있을 것 같다고 하더라고요?
비회원으로 사용하기 기능을 넣어봐도 좋을 것 같아요.

근데 어캐 하는건지는 잘 모름다...ㅋㅎ

**Reply by zangsu** [reply]
데이터는 그냥 로컬 스토리지에 저장하면 될 것 같은데-
카테고리 필터링 하고, 페이징 하고 하는 로직이 전부 JS 에 있어야 하는게 아닌가? 라는 생각이 드네용...

**Reply by Jaymyong66** [reply]
똑같은 기능이 모지?!

**Reply by zangsu** [reply]
지금 구현된 기능! (코드 관리하고 필터링 하고 페이징 하고 ,,,)
이걸 서버와 통신하는 대신 로컬 스토리지를 사용하면 로그인이 없어도 가능할 것 같다는 의견이었습니다. 
다만 서버의 기능을 제공받지 못하는 순간 프론트 업무가 엄청 생길것 같다는 그런 우려였습니다.

**Reply by Jaymyong66** [reply]
오,, 그럼 로컬 스토리지가 비워지면 다 날라가나욥?
아니면 이 이야기인가??
-> 필터링, 페이징 같은건 서버에서 모든 데이터를 다 주면 프론트에서 관리가 가능하지만, 100개 중에 20개씩 준다면 그 20개로 필터링이나 정렬을 해도 나머지 80개가 아닌 데이터로 하기에 정확하지 않은 결과가 온다.

**Reply by zangsu** [reply]
> 오,, 그럼 로컬 스토리지가 비워지면 다 날라가나욥?

이게 맞습니다!
로그인을 하지 않고도 기능을 제공할 수 있는 방법으로는 고려해볼만한 선택지 같아요.
그런데 구체화 하면 할수록 기간 내에 끝내기 힘들 것 같긴.. 합니닷.. ㅎㅎ;

**Comment by jminkkk** [comment]
### 런칭데이 시연에 관하여

우선, 런칭데이에서 내 템플릿이 아닌, 다른 사람의 템플릿을 조회해야 하는 등의 기능이 꼭 필요할까요?
또는 런칭데이에서 시연이 꼭 회원가입인 서비스의 시작부터 흘러가야 할까요?
마지막으로 런칭데이에서 우리의 서비스를 경험하려는 사람은 어떤 것을 보려고 할까요?

런칭데이에는 우리 서비스가 어떤 서비스인지를 어필하는 것이 중요하다고 생각해요.
따라서 베스트 프렉티스 즉, 이미 템플릿을 잘 사용하고 있는 테스트 계정을 제공하는 것이 사용자에게 우리 서비스가 어떤 서비스인지를 알려줄 수 있을 것 같아요.

자신만의 템플릿을 관리하는 방법에도 다양할 것이라고 생각해요. 지금의 카테고리나 태그를 사용해서 말이죠.
예시로 이전의 데모데이에서 켬미는 미션별로 카테고리를 작성하였지만 저는 기능별로 관리할 것 같아요.

여러개의 계정이 다른 분류의 카테고리를 사용해서 템플릿을 잘 관리하고 그 모습을 보여준다면, 저는 충분히 매력적으로 다가올 것 같아요. (관리에 대한 페인포인트를 느꼈던 사람으로써)

다시 말하지만 런칭데이에 시연을 경험하는 사람이 직접 계정을 만들지 않고도 충분히 우리 서비스를 보여줄 수 있다고 생각해요.
다른 느낌으로 관리하는 여러개의 계정을 제시해준다면 충분히 볼거리가 많고 매력을 끌 수 있을 것이라고 생각합니다.

그 이후에 서비스 실사용까지 이어지면 베스트구요.

---

### 비회원 기능

> 비회원도 둘러볼 수 있는 페이지나 기능이 있어야하고, 이후 회원가입을 결정하도록 하면 좋지 않을까?
게다가 런칭데이다! 많은 유저가 쉽게 사용해 볼 수 있어야 한다고 생각한다.

런칭데이와 별개로, 이 부분에 대해서는 매우 동의해요. 
이전에 1박 2일 서비스 기획 캠프? 기획톤?에 참여한 적이 있었는데, 사용자들이 로그인을 꺼려하는 경향이 있어 로그인은 최대한 보류하게 하는 것을 추천한다고 하더라구요.

그런데 기존 페르소나와 아예 방향이 다른 것 같아서 조심스럽기도 합니다. 이야기는 꾸준히 나왔었지만요~
템플릿 관리를 이용하지만 사실 템플릿 관리를 돕는 기능은 아닌 거니까요...??

오히려 템플릿 관리를 이용한 기능을 추가하는 것 말고, 아예 템플릿 대결(켬미말에서 착안) 처럼 사용자가 자신의 템플릿 관리와 별개로 둘 수 있는 새로운 기능도 재미있을 것 같네요.

**Reply by Jaymyong66** [reply]
명확하고 좋은 의견 칭찬해~

**Comment by Hain-tain** [comment]
### 비회원도 사용할 수 있는 기능 
- 탐색하기 탭 꼭 생기면 좋겠다는 생각은 있습니다.
- 안될경우, 우리가 미리 크루들 이름으로 회원가입을 해두는 것은 어때요??

### 로그인 플로우 개선
- 새로운 페이지를 만드는 것이 좋을 것 같아요! 지금은 냅다 로그인 페이지로 리다이렉트 시키고 있는데, 그냥 새로운 페이지를 보여주고, 그 페이지에서  '로그인하러 가가' 또는 '회원가입하러가기'를 눌러 우리 유저가 될 수 있게 하는 것도 좋을 것 같아요!

**Reply by kyum-q** [reply]
크루들이 저장해보려고 해도 실제 사용할 때 의미 없는 데이터를 넣기 때문에 좋지 안은 경험이라고 생각해요
크루들에게 어필할 거는 '써봐' 가 아니라 '다른 사람들은 이렇게 잘 정리해줬구나' 를 볼 수 있다는 것이 좋은 경험이라고 생각합니다.

탐색하기에 좋은 대안은 아니라고 생각합니다 👍
